### PR TITLE
Add allocation strategies with sortable warehouses

### PR DESCRIPTION
**Pull request by orzechdev**
This changes channel page with:
- added allocation strategy card,
- made warehouses sortable.

⚠️ Needs to be merged together with core https://github.com/saleor/saleor/pull/10416

**PR intended to be tested with API branch:** sorting-warehouses-within-shipping-zone

### Screenshots

Screenshot 2022-08-25 at 13 27 38
Screenshot 2022-08-25 at 13 28 23

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7940 Add allocation strategy](https://app.clickup.com/t/2549495/SALEOR-7940)

File: src/channels/components/AssignmentList/AssignmentList.tsx [hunk 1]
```diff
@@ -17,21 +19,46 @@ const messages = defineMessages({
...
 const AssignmentList: React.FC<AssignmentListProps> = props => {
-  const { items, itemsName, totalCount = 0, removeItem } = props;
+  const { items, itemsName, totalCount = 0, removeItem, reorderItem } = props;
 
   const intl = useIntl();
   const classes = useStyles();
   const expanderClasses = useExpanderStyles();
 
+  const handleSortStart = () => {
+    document.body.classList.add(classes.grabbing);
```
Comment (dominik-zeglen) [inline src/channels/components/AssignmentList/AssignmentList.tsx:29]: `react-sortable-hoc` really doesn't add `cursor: pointer` to body?

File: src/channels/components/ChannelForm/ChannelForm.tsx [hunk 2]
```diff
@@ -31,7 +39,10 @@ export interface FormData {
   shippingZonesIdsToRemove: string[];
   warehousesIdsToAdd: string[];
   warehousesIdsToRemove: string[];
+  shippingZonesToDisplay: ChannelShippingZones;
+  warehousesToDisplay: ChannelWarehouses;
   defaultCountry: CountryCode;
+  stockSettings: StockSettingsInput;
```
Comment (dominik-zeglen) [inline src/channels/components/ChannelForm/ChannelForm.tsx:45]: How about `FormData extends StockSettingsInput`?
It'll make form data flat here so we don't have to make weird event spreads ...

File: src/channels/components/ChannelAllocationStrategy/ChannelAllocationStrategy.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              target: {
+                ...event.target,
+                value: {
+                  ...stockSettings,
+                  allocationStrategy: event.target.value,
+                },
+                name: "stockSettings",
+              },
+            })
+          }
```
Comment (dominik-zeglen) [inline src/channels/components/ChannelAllocationStrategy/ChannelAllocationStrategy.tsx:107]: ... like here

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/channels/components/AssignmentList/AssignmentList.tsx (hunk 1 above)
Comment (orzechdev) [reply src/channels/components/AssignmentList/AssignmentList.tsx:29]: Yes, it is missing that

File: src/channels/components/ChannelForm/ChannelForm.tsx (hunk 2 above)
Comment (orzechdev) [reply src/channels/components/ChannelForm/ChannelForm.tsx]: Great idea, fixed

**Comment by timuric** [comment]
Looking good to me

> Select warehouses that will be used in this channel (warehouses can be assigned in multiple channels).

Small copy improvement:

_Assign and sort warehouses that will be used in this channel (warehouses can be assigned in multiple channels)._

**Review by Droniu — APPROVED** [review]

**Review by timuric — APPROVED** [review]

**Review by witoszekdev — APPROVED** [review]

**Review by witoszekdev — APPROVED** [review]

**Pull request merged**